### ISSUE TITLE
The effectiveness of sortagrad

### ISSUE THREAD

**Issue by jinserk** [SeanNaren/deepspeech.pytorch#165]
Hi all,
I have seen that the recent update contains revised bucketing as batch sampler. The basic assumption of this modification is based on the sortagrad is a baseline. However, I couldn't see any effectiveness of the sortagrad from my trials. Could you?
I've slightly changed the model by replacing ReLU in CNN part to [Swish](https://arxiv.org/abs/1710.05941) and using ADAM optimizer with fully shuffled data at every epoches, and this setting gave me the best performance so far. I think the command line option we need is not to choose shuffling only, but to choose sortagrad or not. I'd like to know your opinion about this. Thank you!

**Comment by SeanNaren** [comment]
These are some nice tests, glad swish actually helped out! And that sounds like a fairly straight forward param to add

**Comment by jinserk** [comment]
And I'm curious that if we control the frequency of input data according to its loss value. Intuitively more frequent training of high-loss input, better regularized training result with proper augmentation. But still I don't have a clear picture to implement this..

**Comment by SeanNaren** [comment]
I'll close this issue now, but after a long time since this ticket has been around I haven't seen sortagrad take off or be used. It definitely isn't needed for large scale training and may have been an earlier regularisation technique, but isn't needed per say!
